feat: Add TypeScript variations to meteor create command

jeetburman opened the pull request:
- Added TypeScript template variants for Apollo, Blaze, Vue, Solid, and Svelte
- Updated CLI parser to support --typescript flag with framework flags
- Added shorthand flags: --apollo-typescript, --blaze-typescript, --vue-typescript, --solid-typescript, --svelte-typescript
- Each template includes tsconfig.json, renamed .js to .ts files, and proper TypeScript dependencies
- Updated AVAILABLE_SKELETONS and SKELETON_INFO in commands.js

Enables users to create TypeScript projects with:
  meteor create myapp --vue --typescript meteor create myapp --apollo-typescript

Fixes #12792

jeetburman committed:
feat: Add TypeScript variations to meteor create command

- Added TypeScript template variants for Apollo, Blaze, Vue, Solid, and Svelte
- Updated CLI parser to support --typescript flag with framework flags
- Added shorthand flags: --apollo-typescript, --blaze-typescript, --vue-typescript, --solid-typescript, --svelte-typescript
- Each template includes tsconfig.json, renamed .js to .ts files, and proper TypeScript dependencies
- Updated AVAILABLE_SKELETONS and SKELETON_INFO in commands.js

Enables users to create TypeScript projects with:
  meteor create myapp --vue --typescript
  meteor create myapp --apollo-typescript

Fixes #12792

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/meteor/meteor?pullRequest=13984) <br/>All committers have signed the CLA.

italojs commented on tools/static-assets/skel-apollo-typescript/imports/ui/App.tsx:
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { InMemoryCache, ApolloProvider, ApolloClient, ApolloLink, HttpLink } from '@apollo/client';
+// import { MeteorAccountsLink } from 'meteor/apollo'
```
Could we remove this comment, please?

italojs commented on tools/static-assets/skel-apollo-typescript/server/main.ts:
```diff
@@ -0,0 +1,39 @@
+import { LinksCollection } from '/imports/api/links';
+import { startApolloServer } from './apollo';
+
+async function insertLink({ title, url }) {
```
missing more types in the hole PR, at least in the server

here we could have something like
```
interface LinkInput {
  title: string;
  url: string;
}

async function insertLink({ title, url }: LinkInput): Promise<void> {
  await LinksCollection.insertAsync({ title, url, createdAt: new Date() });
}
```

italojs commented on tools/cli/commands.js:
```diff
@@ -690,57 +690,73 @@ function getExamplesJSON(){
+  "apollo-typescript",
   "bare",
   "blaze",
+  "blaze-typescript",
```
IMO, for DX propose we should have a separate command like `--blaze --<anything> --typescript`

jeetburman committed:
fix: Add proper TypeScript types to server files

- Added interface definitions for function parameters
- Added return type annotations (Promise<void>, void)
- Uncommented MeteorAccountsLink import in Apollo template
- Improved type safety across all TypeScript templates

Addresses maintainer feedback on #12792

Grubba27 commented on tools/cli/commands.js:
```diff
@@ -915,11 +931,26 @@ main.registerCommand({
-      const skeleton = skeletonExplicitOption || DEFAULT_SKELETON;
-
+      
+      // Check for TypeScript + Framework combinations first
+      let skeleton;
+      if (options.typescript && options.vue) {
+        skeleton = 'vue-typescript';
+      } else if (options.typescript && options.apollo) {
+        skeleton = 'apollo-typescript';
+      } else if (options.typescript && options.blaze) {
+        skeleton = 'blaze-typescript';
+      } else if (options.typescript && options.solid) {
+        skeleton = 'solid-typescript';
+      } else if (options.typescript && options.svelte) {
+        skeleton = 'svelte-typescript';
+      } else {
+        // Check for explicit skeleton option or use default
+        const skeletonExplicitOption =
+          AVAILABLE_SKELETONS.find(skeleton => !!options[skeleton]);
+        skeleton = skeletonExplicitOption || DEFAULT_SKELETON;
+      }
```
Would passing `--typescript` still create a React TypeScript app?

Grubba27 replied:
I would prefer something more in a functional style:

```suggestion
const skeleton = (() => 
       if (options.typescript) {
          return AVAILABLE_TYPESCRIPT_SKELETONS.find(skeleton => !!options[skeleton]);
       } else { AVAILABLE_SKELETONS.find(skeleton => !!options[skeleton]); }
      )()
```

italojs commented:
@jeetburman , will you work on it after the Hackerranktoberfest?

jeetburman commented:
> @jeetburman , will you work on it after the Hackerranktoberfest?

I am willing to work on this. I just couldnt for the past few weeks. I will amend the PR as per the review provided shortlly.

jeetburman commented:
> @jeetburman , will you work on it after the Hackerranktoberfest?

Also, I had a question...
Does this organization participate or is planning to participate in Google Summer of Code?
